rfcs/0001-generic-launch-receipting-pattern.md

```diff
@@ -0,0 +1,195 @@
+
+To comply with the format, RM systems must include a `version` property in the JWT payload provided to eQ. This enables eQ to determine the payload's version and apply different validators and parsers. The option of doing this via the existence of specific keys by one survey type or another was avoided to make the version explicit and make future payload versions easier to manage. To avoid modifying the existing payload generators, this new `version` field is unnecessary for the current v1 format. If the property is not present, eQ will fall back to the v1 model. From version 2 onwards, however, this attribute is necessary.
+
+The v2 design removes all top-level properties not shared by all survey types or RM systems from the launch token. The only top-level properties that will be required are:
+
+- iat
+- exp
+- jti
+- account_service_url
+- case_id
+- collection_exercise_id
+- response_id
+- tx_id
+- version
+- either schema_url or schema_name
+
+#### What about additional data about the survey?
+
+RM systems will almost always want to provide extra survey data, such as data that eQ needs to pipe (render) or other metadata that must be passed to downstream systems. A new property called `survey metadata` has been introduced to facilitate this. This property is an object with two sub-properties. First, there's the `data` (survey_metadata.data) property, which should contain any additional survey-specific metadata required by eQ or downstream.
+
+For example:
+
+```json  
+"survey_metadata": {  
+  "data": {  
+    "case_ref": "1000000000000001",  
+    "case_type": "B",  
+    "form_type": "0253",  
+    "period_id": "202101",  
+    "period_str": "January 2021",  
+    "ref_p_end_date": "2021-06-01",  
+    "ref_p_start_date": "2021-01-01",  
+    "ru_name": "ACME T&T Limited",  
+    "ru_ref": "49900000001A",  
+    "user_id": "64389274239"  
+  }  
+}  
+```  
+
+The second property is `recepting_keys` (survey_metadata.recepting_keys), an array of key names from the `survey metadata.data` field that is necessary for receipting. The `recepting_keys` property is optional and is only required for RM systems that require more data in addition to the existing receipting data, which uses 'tx id' and 'case id'. The reason `recepting_keys` is required is to prevent hard-coding the receipting requirements of different survey types in eQ Runner, which means that RM systems can modify these receipting requirements in the future without requiring application changes in Runner. This is not required for business surveys, but RH will require it for social surveys because each `case_id` can have numerous responses. An RH social survey JWT `survey_metadata` payload would resemble something like this:
+
+```json  
+"survey_metadata": {  
+  "data": {  
+    "case_ref": "1000000000000001",  
+    "case_type": "B",  
+    "questionnaire_id": "bdf7dff2-1d73-4b97-bd2d-91f2e53160b9"  
+  },  
+  "receipting_keys": [  
+    "questionnaire_id"  
```

Comment:
will EQ do any validation here i.e. validate that the receipting_keys values exists as an item in the survey_metadata block? or is it assumed validated before launch

Reply:
eQ will do some basic validation to ensure keys specified in `receipting_keys` exist in `survey_metadata.data` block. We will only check that the key exists and that is not 0 length, but we will not be doing any further validation on the value. https://github.com/ONSdigital/ons-schema-definitions/blob/generic-launch-and-recepiting-pattern/docs/rm_to_eq_runner_payload_v2.rst#survey-metadata-fields

Do note, eQ may additionally validate the values in `survey_metadata.data` depending on the outcome of https://github.com/ONSdigital/ons-schema-definitions/blob/generic-launch-and-recepiting-pattern/rfcs/0001-generic-launch-receipting-pattern.md#unresolved-questions

Reply:
👍

Reply:
👍